spec/javascripts/unit/core/channels/presence_channel_spec.js

```diff
@@ -156,6 +164,43 @@ describe("PresenceChannel", function() {
         });
       });
 
+      describe("pusher_internal:subscription_succeded but subscription cancelled", function() {
+        it("should not emit pusher:subscription_succeded", function() {
+          var callback = jasmine.createSpy("callback");
+          channel.bind("pusher:subscription_succeeded", callback);
+
+          channel.cancelSubscription();
+          channel.handleEvent("pusher_internal:subscription_succeeded", "123");
+
+          expect(callback).not.toHaveBeenCalled();
+        });
+
+        it("should set #subscribed to true", function() {
+          channel.cancelSubscription();
+          channel.bind(function() {
+            expect(channel.subscribed).toEqual(true);
+          });
+          channel.handleEvent("pusher_internal:subscription_succeeded");
+        });
+
+        it("should set #subscriptionPending to false", function() {
+          channel.cancelSubscription();
+          channel.bind(function() {
+            expect(channel.subscriptionPending).toEqual(true);
+          });
+          channel.handleEvent("pusher_internal:subscription_succeeded");
```

Comment:
I copied the format of these from one of the existing tests and on closer inspection I'm not really sure what they are doing. I'm not sure what binding without specifying an event even does... will rewrite these so they make more sense to me.